File: lecture/02_Sort1/BubbleSort.java
import java.util.Scanner;
class BubbleSort{
    public static void main(String args[]){
	Scanner sc = new Scanner(System.in);
	int x[];
	int num = sc.nextInt();
	try{
	    x = new int[num];
	    for (int i = 0; i < num; i++) 
		x[i] = sc.nextInt();
	    int t = bubbleSort(x, num);
	    System.out.println(t);
	}catch(Exception e){
	    e.printStackTrace();
	}	
    }
    public static int bubbleSort(int a[],int n){
	int cnt = 0;
	for (int i = 0; i < n-1; i++) {
	    for (int j = i+1; j < n; j++) {
		if(a[i]>a[j]){
		    swap(a,i,j);
		    cnt++;
		}
	    }
	}
	for (int i = 0; i < n; i++) 
	    System.out.print(a[i]+" ");	
	System.out.println();
	return cnt;
    }
    public static void swap(int a[], int idx1, int idx2){
	int tmp = a[idx1];	a[idx1] = a[idx2];	a[idx2] = tmp;
    }
}

File: lecture/02_Sort1/BubbleSort.c
#include <stdio.h>

void swap(int *a, int *b);

int main(void) {
  int n, i, j, cnt = 0;
  scanf("%d", &n);

  int a[n]; 
  for (i = 0; i < n; i++) {
    scanf("%d", &a[i]);
  }

  for (i = 0; i < n-1; i++) {
    for (j = n-1; j > i; j--) {
      if (a[j] < a[j-1]) {
	swap(&a[j], &a[j-1]);
	cnt++;
      }
    }
  }

  for (i=0; i<n;i++) {
    if (i == n - 1)
      printf("%d\n",a[i]);
    else
      printf("%d ", a[i]);
  }
  printf("%d\n",cnt);
  return 0;
}

void swap(int *a, int *b) {
  int tmp = *a;
  *a = *b;
  *b = tmp;
}
